plugins/nominal-connection-checker/src/hierarchy_validation.js

```diff
@@ -228,6 +229,48 @@ function checkSupersDefined(hierarchyDef) {
   }
 }
 
+/**
+ * Checks the hierarchy def for any types which include the same type multiple
+ * times in their fulfills array.
+ * @param {!Object} hierarchyDef The definition of the type hierarchy.
+ */
+function checkDupeSupers(hierarchyDef) {
+  const errorMsg = 'The type %s fulfills the type %s multiple times.';
+
+  for (const type of Object.keys(hierarchyDef)) {
+    const typeInfo = hierarchyDef[type];
+    if (!typeInfo.fulfills || !Array.isArray(typeInfo.fulfills)) {
+      continue;
+    }
+    const superOccurrences = new Map();
+    const superNames = new Map();
```

Comment:
You could combine these maps by having the value for the map be an object like:
```
{
  count: {number}
  caseSensitiveName: {string}
}
```
You would only have to check for existence in the map once inside of the loop.